### PR TITLE
WIP: Submission pipeline

### PR DESCRIPTION
**Pull request by jatinkatyal13**

### CODE REVIEW

File: src/tasks/index.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import {exec, mkdir, rm} from 'shelljs'
+import * as path from 'path'
+
+import RunScenario from './run'
+import SubmissionScenario from './submission'
+import { RunJob, SubmitJob } from "./job";
+
+export function execute(job: RunJob): Promise<RunResult>
+export function execute(job: SubmitJob): Promise<SubmissionResult>
+export async function execute (job) {
```
Comment (championswimmer) [inline src/tasks/index.ts:12]: `export async function execute (job: Job) {`

File: src/tasks/job.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+type JobConstructorOpts = {
```
Comment (championswimmer) [inline src/tasks/job.ts:1]: make this an interface

File: src/tasks/job.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  id: number
+  source: string
+  lang: string
+  timelimit?: number
+} 
+type RunJobConstructorOpts = JobConstructorOpts & {
```
Comment (championswimmer) [inline src/tasks/job.ts:7]: then this can be an another Interface that extends the Job interface

File: src/types/result.d.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+export type Result = {
+  id: number,
+  stderr: string
+}
+
+export type TestcaseResult = {
```
Comment (championswimmer) [inline src/types/result.d.ts:6]: use interfaces and OOP-style extension here too

Comment (jatinkatyal13) [reply src/types/result.d.ts]: done

File: src/tasks/job.ts (hunk 3 above)
Comment (jatinkatyal13) [reply src/tasks/job.ts]: done

**Pull request merged**